### PR TITLE
Diary Local Storage and refactoring

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR:

- Refactoring
- Diary localStorage

FYI @akintner @case-eee @icorson3

### CODE REVIEW

File: lib/foods.js [hunk 1]
```diff
@@ -1,82 +1,98 @@
...
-      $('#food-name, #calorie-count').val("");
-      storeFood(foodName, foodCalories);
+      storeAndPrependFood(foodName, foodCalories)
     }
   });
+}
+
+function storeAndPrependFood(food, calories){
+  prependFood(food, calories)
+  $('#food-name, #calorie-count').val("");
```
Comment (case-eee) [inline lib/foods.js]: this could be a function called `clearNameAndCount`?

Comment (Dpalazzari) [inline lib/foods.js]: I have rectified this inequity 👍 @case-eee

Comment (case-eee) [inline lib/foods.js]: woohooo @Dpalazzari great work!

**Comment by Dpalazzari** [comment]
We are done with this project forever @case-eee

**Comment by akintner** [comment]
We just worked on this for 4 hours together and spun in circles with it, so we're committing and letting it be. Doesn't work yet, though what we had would work for one round of localStorage setting, then break on the next item added to a food table. No idea. We burned ourselves out on it, though. It's probably something small that we just went down too far in the rabbit hole to see. Every time we tried something new, another thing broke in a different place. C'est la vie. Here's what we've got.

**Pull request merged**